validator/sawtooth_validator/networking/interconnect.py

```diff
@@ -69,12 +75,44 @@ def __init__(self, connection, address, futures, identity=None,
         self._secured = secured
         self._server_public_key = server_public_key
         self._server_private_key = server_private_key
+        self._heartbeat = heartbeat
+        self._heartbeat_interval = heartbeat_interval
+
         self._event_loop = None
         self._context = None
         self._recv_queue = None
         self._socket = None
         self._condition = Condition()
 
+        self._connected_identities = {}
+
+    @asyncio.coroutine
+    def _send_heartbeat(self):
+        with self._condition:
+            self._condition.wait_for(lambda: self._socket is not None)
+
+        ping = PingRequest()
+
+        while True:
```

Comment:
It's an asyncio coroutine. It yields until the event loop stops servicing it.

There is a broader question here about cleanup of connections in the dict once they are no longer live. This PR doesn't attempt to address that. Detecting stale connections and doing appropriate connection cleanup will occur in a future PR.